src/presolve.jl

```diff
@@ -376,6 +385,9 @@ Perform pre-solve.
 """
 function presolve!(ps::PresolveData{T}) where {T}
 
+    # Ensure the bounds of integer variables are integers.
+    ensure_integer_bounds!(ps)
```

Comment:
This should go in the main loop. For instance just before fixed variables are removed.

Reply:
Should it be better just before the line @_return_if_inferred bounds_consistency_checks!(ps)?

Reply:
These are fairly cheap operations, so the exact order shouldn't matter too much (worst case we do another loop).